Documentation/ceph-csi-troubleshooting.md

```diff
@@ -0,0 +1,388 @@
+
+If any issue exists in attaching the PVC to the application pod check logs from driver-registrar
+sidecar container in plugin pod where your application pod is scheduled.
+
+```console
+$ kubectl logs po/csi-rbdplugin-vh8d5 -c driver-registrar
+I0120 12:28:34.231761  124018 main.go:112] Version: v2.0.1
+I0120 12:28:34.233910  124018 connection.go:151] Connecting to unix:///csi/csi.sock
+I0120 12:28:35.242469  124018 node_register.go:55] Starting Registration Server at: /registration/rook-ceph.rbd.csi.ceph.com-reg.sock
+I0120 12:28:35.243364  124018 node_register.go:64] Registration Server started at: /registration/rook-ceph.rbd.csi.ceph.com-reg.sock
+I0120 12:28:35.243673  124018 node_register.go:86] Skipping healthz server because port set to: 0
+I0120 12:28:36.318482  124018 main.go:79] Received GetInfo call: &InfoRequest{}
+I0120 12:28:37.455211  124018 main.go:89] Received NotifyRegistrationStatus call: &RegistrationStatus{PluginRegistered:true,Error:,}
+E0121 05:19:28.658390  124018 connection.go:129] Lost connection to unix:///csi/csi.sock.
+E0125 07:11:42.926133  124018 connection.go:129] Lost connection to unix:///csi/csi.sock.
+```
+
+You should see the response `RegistrationStatus{PluginRegistered:true,Error:,}` in the logs to
+confirm that plugin is registered with kubelet.
+
+If you see a driver not found an error in the application pod describe output.
+Restarting the `csi-xxxxplugin-xxx` pod on the node may help.
+
+## Volume Attachment
+
+Each provisioner pod also has a sidecar container called `csi-attacher`.
+
+### csi-attacher
+
+The external-attacher is a sidecar container that attaches volumes to nodes by calling `ControllerPublish` and
+`ControllerUnpublish` functions of CSI drivers. It is necessary because the internal Attach/Detach controller
+running in Kubernetes controller-manager does not have any direct interfaces to CSI drivers. More details can
+be found [here](https://github.com/kubernetes-csi/external-attacher).
+
+If any issue exists in attaching the PVC to the application pod first check the volumettachment object created
+and also log from csi-attacher sidecar container in provisioner pod.
+
+```console
+$ kubectl get volumeattachment
+NAME                                                                   ATTACHER                        PV                                         NODE       ATTACHED   AGE
+csi-75903d8a902744853900d188f12137ea1cafb6c6f922ebc1c116fd58e950fc92   rook-ceph.cephfs.csi.ceph.com   pvc-5c547d2a-fdb8-4cb2-b7fe-e0f30b88d454   minikube   true       4m26s
+
+$ kubectl logs po/csi-rbdplugin-provisioner-d857bfb5f-ddctl -c csi-attacher
+```
+
+## CephFS Stale operations
+
+Check for any stale mount commands on the `csi-cephfsplugin-xxxx` pod on the node where your application pod is scheduled.
+
+You need to exec in the `csi-cephfsplugin-xxxx` pod and grep for stale mount operators.
+
+Identify the `csi-cephfsplugin-xxxx` pod running on the node where your application is scheduled with
```

Comment:
Use the deployment instead

Reply:
Was this actually resolved? Why not use `kubectl -n rook-ceph exec -it daemonset/csi-cephfsplugin -c csi-cephfsplugin -- bash`?

Reply:
We need to find the plugin on the same node where the app pod is mounting the volume, not sure how that command helps.

Conceptually, I think there is also a balance to consider about documenting the approach vs providing exact commands for every operation. If we're too detailed, it's not necessarily helpful to finding issues.

Reply:
@BlaineEXE see [this](https://github.com/rook/rook/pull/7157#discussion_r571802778) for reason behind not using deployment.